### ISSUE TITLE
Channel Name not recognised

### ISSUE THREAD

**Issue by saschafoerster** [AddReminders/Reminders-in-RocketChat#10]
We had channel names with underscore which where changed by RocketChat to -, until we manually allowed special characters for channel names. Now I have to impression, that the channel name is shown with _ but it can only be addressed with RemindMe, when we use the old and first channel name with -. So the idea: could the channel name be autocompleted?
Because right now, the new channel name with _ is not recognised as valid by the app, and when we use the old name (probably main name in the database but not shown), we don't get reminders in them, even though the channel name is accepted.

**Comment by saschafoerster** [comment]
I can confirm (with RocketChat 5.0.0 and AddReminders 2.3.0
- Channels with regular name, that has not been changed: Regular Reminders work. 
- Channels with special letters allowed (like _ ) and that have been changed after created: Reminders don't work.

**Comment by murtaza98** [comment]
Sorry for the delayed response here :disappointed: This seems to be happening because the app is storing the room name which is subject to change once the room of the name is changed. I'll fix this in the upcoming release (end of August) by migrating to use `roomId` instead of `roomName`. 

Thanks for reporting this issue :raised_hands:

**Comment by saschafoerster** [comment]
@murtaza98 To be clear: we created the channel, we then allowed underscores in the options, we then changed the name of the channel and only after all of this we started using this app. So the reminder is created AFTER the room name has changed, not the other way round (as your new title suggests). 

I had several issues because of this already, also with weebhooks and integrations. I sometimes have to guess the "first name" of a room and can't use the room name as I can see it. So I though already there must be some archive our two separat tables with names, one with the old name, one with the changed name. I am not sure if it is only roomID and roomName or if there is a third kind of "roomNameFirst" or "roomNameWithoutSpecialSigns". ;)

**Comment by murtaza98** [comment]
Sorry, my bad! Let me handle that case separately. 

Wrt this issue, can you please share some steps to reproduce this issue?

I've already tried the following but wasn't able to replicate the issue
1. Created a channel with the name "test_1"
2. Enabled the following setting to allow special chars within name.
	![image](https://user-images.githubusercontent.com/34130764/182642890-37e0156b-da58-4672-a9d9-c7bfaacff9fc.png)
3. Renamed the channel name to "test_1_updated.name"
4. Created a reminder for the channel using [the "Add Reminder" button
](https://add-reminders.gitbook.io/docs/guides/setting-up-reminders/set-a-reminder-for-someone-else-or-a-channel#option-2-using-message-input-action-button). This reminder was properly setup and sent on the channel
![image](https://user-images.githubusercontent.com/34130764/182643596-f1d61c7b-c87e-452b-9285-bbb3321e0503.png)

**Comment by saschafoerster** [comment]
Hm, it's just a guess: I have found in mongodb two names for a room: name and fname

"fname" seems to me to be the roomname that is shown visibly for the user. "fname" is also the name of the room after I changed it. 
"name" is the name, when I first created to room.

In our instance we have two rooms, where I tested repeating reminders. It works in #z_rocketchat, but it doesn't work in #_bonndigital_teammett

It seems also that name and fname support underscores if it is activated. So I guess for testing you have to create a room without underscore, when it is not allowed, and then allow underscores and change the roomname to something with underscore to create the same database entry. 

We have in mondodb:

```
name
bonndigital-teammett
fname
_bonndigital_teammett
```
and
```
name
z_rocketchat
fname
z_rocketchat
```

In the screenshot you can also see, that for the room name you seem to use "name" and not "fname" (when I create a simple reminder from /remind): you can see the old name of the room "bonndigital-teammett" and not "_bonndigital_teammett" as it should be. 

<img width="593" alt="Bildschirmfoto 2022-08-07 um 10 53 00" src="https://user-images.githubusercontent.com/1153741/183283241-69145af7-20dd-4da9-9806-12a2170835db.png">

**Comment by saschafoerster** [comment]
I wonder also in your replication, why is there this difference between the room names: 
In step 3 you renamed the channel name to "test_1_updated.name" with a point. 
But the remind-button then creates a reminder for "test_1_updated-name" with a minus. This is not really coherent, how can I guess the name change there? 

So maybe for replication could you maybe try this: 

1. Created a channel with the name "test-1" (please use minus and not underscore here!) 
2. Enabled the setting to allow special chars within name after that. 
3. Change the channel name to "_test_1" (with some underscore in the beginning of the room name)
4. Create a reminder with /remind where you manually add the room name "_test_1" 

I am also unclear if I should add # in front of the room name or not.

**Comment by saschafoerster** [comment]
Oh, and it seems to make a difference, if the reminders is a repeating reminder. So please try it with a reminder that comes back daily. The single reminder worked for me with the channel name, but the repeating reminder didn't.

**Comment by murtaza98** [comment]
Thanks for the brief explanation @saschafoerster Thanks to it I was finally able to reproduce this issue locally. 
![image](https://user-images.githubusercontent.com/34130764/183893532-c94c7fd4-973e-45e0-92b7-0bfb39701aa2.png)

As you mentioned the issue is because the app was using `name` property on the UI instead of the `fname` property. I'll rectify this in the next release of August by using `fname` prop everywhere on UI where we show the room name. However do note that while creating the reminder, you'd still need to use `name` & NOT `fname` - this is because on the backend we always deal with `name` & never with `fname`. So for simplicity, I'd recommend using [the "Add Reminder" button](https://add-reminders.gitbook.io/docs/guides/setting-up-reminders/set-a-reminder-for-someone-else-or-a-channel#option-2-using-message-input-action-button) to create a reminder for a room - it would autofill the name prop.

**Comment by saschafoerster** [comment]
I have the same issue with integrations of RocketChat: I have to guess the name, because it is not visible anywhere (beside inside of the database). So many times I didn't remember the first "name" of a room. Why is it that fname can't be used? Or at least a system that shows all room names with fname and name, so I can select it from existing rooms? Guessing the "name" is not really user friendly (even though there is the workaround with the button)

**Comment by saschafoerster** [comment]
But I am happy that I could create a reproducible report and thank you so much for developing this really useful add on. Should be simply integrated into RocketChat, I think.

**Comment by murtaza98** [comment]
> I have the same issue with integrations of RocketChat: I have to guess the name, because it is not visible anywhere (beside inside of the database). So many times I didn't remember the first "name" of a room. Why is it that fname can't be used? Or at least a system that shows all room names with fname and name, so I can select it from existing rooms? Guessing the "name" is not really user friendly (even though there is the workaround with the button)

If you're using the browser, then you can get the `name` from the Url of the webpage. Like in my above example, my URL for the room with `fname=.test.1` was `http://localhost:3000/group/test-1` - so from this you get the `name` as `test-1`

**Comment by saschafoerster** [comment]
Ah, good to know! That saves me a lot of time. I had to look into the database until now. :D

**Comment by murtaza98** [comment]
Hey, Just an FYI: This was fixed with https://github.com/AddReminders/Reminders-in-RocketChat/releases/tag/2.4.0 and has been submitted to the RocketChat marketplace. But the review is still pending by the team so it's now publically available. However, if you don't wish to wait until it is available on the marketplace, then you may install it manually on your server :smile: (Info in above link)

Thanks for reporting this issue and helping improve this app :people_hugging:

**Comment by murtaza98** [comment]
Released with Version 2.7 on marketplace. Version also available here for offline install -> https://github.com/AddReminders/Reminders-in-RocketChat/releases/tag/2.7.0
